ui/admin/app/components/auth-and-model-providers/ProviderIcon.tsx

```diff
@@ -1,31 +1,40 @@
 import { BoxesIcon } from "lucide-react";
+import { useEffect, useState } from "react";
 
 import { AuthProvider, ModelProvider } from "~/lib/model/providers";
 import { cn } from "~/lib/utils";
 
-import {
-	CommonAuthProviderIds,
-	CommonModelProviderIds,
-} from "~/components/auth-and-model-providers/constants";
-
 export function ProviderIcon({
 	provider,
 	size = "md",
 }: {
 	provider: ModelProvider | AuthProvider;
 	size?: "md" | "lg";
 }) {
+	const [isDarkMode, setIsDarkMode] = useState(false);
+
+	useEffect(() => {
+		const darkModeMediaQuery = window.matchMedia(
+			"(prefers-color-scheme: dark)"
+		);
+		setIsDarkMode(darkModeMediaQuery.matches);
+
+		const handleChange = (e: MediaQueryListEvent) => {
+			setIsDarkMode(e.matches);
+		};
+
+		darkModeMediaQuery.addEventListener("change", handleChange);
+		return () => darkModeMediaQuery.removeEventListener("change", handleChange);
+	}, []);
+
```

Comment:
This logic is already abstracted

```suggestion
	const { isDark } = useTheme()

```

Reply:
Cool!